libs/seiscomp/datamodel/utils.cpp

```diff
@@ -379,13 +379,13 @@ bool by_Z_desc_and_code_asc(const ComponentAxis &axis1, const ComponentAxis &axi
 }
 
 
-bool getThreeComponents(ThreeComponents &res, const SensorLocation *loc, const char *streamCode, const Core::Time &time) {
+bool getThreeComponents(ThreeComponents &tc, const SensorLocation *loc, const char *streamCode, const Core::Time &time) {
 	int len = strlen(streamCode);
 
-	res = ThreeComponents();
-
-	std::vector<ComponentAxis> comps;
+	tc = ThreeComponents();
 
+	// First find matching streams based on code and time
+	std::vector<Stream*> matchingStreams;
```

Comment:
Because I prefer to separate the two steps:

1. collect the matching streams and give up if their number is not 3
2. if 3 streams were found, check if these are perpendicular

In fact this could even be done in separate functions.

To me when I debugged the problem with a specific sensor location, the code was initially not easy to understand and the motivation for the change is to improve this. But YMMV of course.

Reply:
As I have showed above, it won't work for all cases. Even if the code might be more easy to understand (which I do not agree with), it removes functionality which I do not prefer. The aim of this PR is to be more tolerant in terms of orientation, but with that change in place we are even less tolerant in some situations and it has actually nothing to do with the topic of this PR. I would like to keep PRs small, simple and focused. Even renaming a variable should be done in a separate commit separated from the actual changes related to this PR.